label_studio/fsm/signals.py

```diff
@@ -0,0 +1,177 @@
+"""
+Django signal handlers for Label Studio core FSM integrations.
+
+These signal handlers provide automatic state transitions for core Label Studio
+workflows. They are designed to:
+1. Work only when enterprise is not present
+2. Respect the same feature flags as enterprise
+3. Provide basic state tracking without interfering with enterprise functionality
+4. Be resilient to errors (never break core functionality)
+"""
+
+import logging
+from typing import Optional
+
+from django.contrib.auth.models import User
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+logger = logging.getLogger(__name__)
+
+
+def get_user_from_instance(instance) -> Optional[User]:
+    """
+    Extract user from model instance using common field patterns.
+
+    Args:
+        instance: Django model instance
+
+    Returns:
+        User instance if found, None otherwise
+    """
+    # Try common user field patterns
+    for field in ['created_by', 'updated_by', 'completed_by', 'user']:
+        if hasattr(instance, field):
+            user = getattr(instance, field)
+            if isinstance(user, User):
+                return user
+    return None
+
+
+@receiver(post_save)
+def handle_model_state_transitions(sender, instance, created, **kwargs):
```

Comment:
I think it would be nice to break this up a bit, from a large if..elif..elif... wrapped in a try-except. How about:

1) implementing the error handling in a decorator
2) implementing a handler for each model as a function, decorated with said decorator
3) using a "registry" mapping each model name to its handler function
4) when we are going to do a transition, pull the function from the registry and call it. Log appropriately if no handler is registered for the model in question

This would make it easier to test each state transition handler in a decoupled fashion, as well as to gradually add new ones and increase their complexity over time

Reply:
Done 👍, uses explicit modelled Transition objects instead.